apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+          }
+        }
+
+        // If we miss any filter group, then we don't show the tutorial.
+        var filterGroupMiss = false;
+
+        for (var filterGroupName in this.props.filters) {
+          var tutorialTags = tutorial["tags_" + filterGroupName];
+          if (tutorialTags && tutorialTags.length > 0) {
+            var tutorialTagsSplit = tutorialTags.split(',');
+
+            // now check all the filter group's tags
+            var filterGroup = this.props.filters[filterGroupName];
+
+            // For this filter group, we've not yet found a matching tag between
+            // user selected otions and tutorial tags.
+            var filterHit = false;
+
+            for (var filterName of filterGroup) {
+              if (tutorialTagsSplit.includes(filterName)) {
+                // The tutorial had a matching tag.
+                filterHit = true;
+              }
+            }
+
+            // The filter group needs at least one user-selected filter to hit
+            // on the tutorial.
+            if (filterGroup.length !== 0 && !filterHit) {
+              filterGroupMiss = true;
+            }
+          }
+        }
+
+        return !filterGroupMiss;
+      }
+
+      return (
+        <div className="col-80" style={{float: 'left'}}>
+          {this.props.tutorials.filter(filterFn, this).map(item => <Tutorial item={item} filters={this.props.filters} key={item.code}/>)}
+        </div>
+      );
+    }
+  });
+
+  const TutorialExplorer = React.createClass({
+    propTypes: {
+      filterGroups: React.PropTypes.array,
+      tutorials: React.PropTypes.array,
+      locale: React.PropTypes.string
+    },
+
+    getInitialState: function () {
+      var filters = {};
+
+      for (let filterGroup of this.props.filterGroups) {
+        filters[filterGroup.name] = [];
+
+      }
+      return {
+        filters: filters
+      };
+    },
+
+    handleUserInput: function (filterGroup, filterEntry, value) {
+      var filterEntryChange = {};
+
+      if (value) {
+        // Add value to end of array.
+        filterEntryChange["$push"] = [filterEntry];
+
+      } else {
+        var itemIndex = this.state.filters[filterGroup].indexOf(filterEntry);
+
+        // Find and remove specific value from array.
+        filterEntryChange["$splice"] = [[itemIndex, 1]];
+      }
+
+      var filterGroupChange = {};
+      filterGroupChange[filterGroup] = filterEntryChange;
+
+      var stateChange = {};
+      stateChange["filters"] = filterGroupChange;
+
+      var newState = update(this.state, stateChange);
+
+      this.setState(newState);
+    },
+
+    render() {
+      return (
+        <div>
+          <FilterSet filterGroups={this.props.filterGroups} onUserInput={this.handleUserInput} selection={this.state.filters}/>
+          <TutorialSet tutorials={this.props.tutorials} filters={this.state.filters} locale={this.props.locale}/>
+        </div>
+      );
+    }
+  });
+
+  window.ReactDOM.render(
```

Comment:
Rather than rendering this in our ctor, I'd prefer to expose a child method, such that we end up doing something like:
```
var manager = new TutorialExporerManager(options);
manager.renderToElement(document.getElementById('tutorials'));